internal/services/appservice/helpers/web_app_slot_schema.go

```diff
@@ -275,6 +275,269 @@ func SiteConfigSchemaLinuxWebAppSlot() *pluginsdk.Schema {
+	ManagedPipelineMode      string                    `tfschema:"managed_pipeline_mode"`
+	RemoteDebugging          bool                      `tfschema:"remote_debugging_enabled"`
+	RemoteDebuggingVersion   string                    `tfschema:"remote_debugging_version"`
+	ScmType                  string                    `tfschema:"scm_type"`
+	Use32BitWorker           bool                      `tfschema:"use_32_bit_worker"`
+	WebSockets               bool                      `tfschema:"websockets_enabled"`
+	FtpsState                string                    `tfschema:"ftps_state"`
+	HealthCheckPath          string                    `tfschema:"health_check_path"`
+	HealthCheckEvictionTime  int                       `tfschema:"health_check_eviction_time_in_min"`
+	WorkerCount              int                       `tfschema:"worker_count"`
+	VirtualApplications      []VirtualApplication      `tfschema:"virtual_application"`
+	MinTlsVersion            string                    `tfschema:"minimum_tls_version"`
+	ScmMinTlsVersion         string                    `tfschema:"scm_minimum_tls_version"`
+	Cors                     []CorsSetting             `tfschema:"cors"`
+	DetailedErrorLogging     bool                      `tfschema:"detailed_error_logging_enabled"`
+	WindowsFxVersion         string                    `tfschema:"windows_fx_version"`
+	VnetRouteAllEnabled      bool                      `tfschema:"vnet_route_all_enabled"`
+}
+
+func SiteConfigSchemaWindowsWebAppSlot() *pluginsdk.Schema {
+	return &pluginsdk.Schema{
+		Type:     pluginsdk.TypeList,
+		Required: true,
+		MaxItems: 1,
+		Elem: &pluginsdk.Resource{
+			Schema: map[string]*pluginsdk.Schema{
+				"always_on": {
+					Type:     pluginsdk.TypeBool,
+					Optional: true,
+					Default:  true,
+				},
+
+				"api_management_api_id": {
+					Type:         pluginsdk.TypeString,
+					Optional:     true,
+					ValidateFunc: apimValidate.ApiID,
+				},
+
+				"api_definition_url": {
+					Type:         pluginsdk.TypeString,
+					Optional:     true,
+					ValidateFunc: validation.IsURLWithHTTPorHTTPS,
+				},
+
+				"application_stack": windowsApplicationStackSchema(),
+
+				"app_command_line": {
+					Type:     pluginsdk.TypeString,
+					Optional: true,
+				},
+
+				"auto_heal_enabled": {
+					Type:     pluginsdk.TypeBool,
+					Optional: true,
+					Default:  false,
+					RequiredWith: []string{
+						"site_config.0.auto_heal_setting",
+					},
+				},
+
+				"auto_heal_setting": autoHealSettingSchemaWindows(),
+
+				"auto_swap_slot_name": {
+					Type:         pluginsdk.TypeString,
+					Optional:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+
+				"container_registry_use_managed_identity": {
+					Type:     pluginsdk.TypeBool,
+					Optional: true,
+					Default:  false,
+				},
+
+				"container_registry_managed_identity_client_id": {
+					Type:         pluginsdk.TypeString,
+					Optional:     true,
+					ValidateFunc: validation.IsUUID,
+				},
```

Comment:
just double-checking this: every other Identity thing in Azure uses the Managed Identity ID - I'm assuming this is an API limitation

Reply:
The API will accept any string, however, it can only actually pull images if the value is a valid and allowed client UUID.